frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -557,3 +569,226 @@ export default function EditorDialog(props: EditorDialogProps) {
+        sx={{
+          bgcolor: 'grey.700',
+          color: 'white',
+          px: 5,
+          '&:hover': { bgcolor: 'grey.600' },
+        }}
+      >
+        {t('translation|Load')}
+      </Button>
+    </Box>
+  );
+
+  const finishLoad = (text: string) => {
+    setCode({ format: 'yaml', code: text });
+    setUploadYaml(false);
+  };
+
+  const UploadYamlFromFilesystem = () => {
+    const [dragOver, setDragOver] = React.useState(false);
+
+    const handleLoadFile = () => {
+      if (!uploadedfile) {
+        return;
+      }
+
+      const reader = new FileReader();
+      reader.onload = e => finishLoad(e.target?.result as string);
+      reader.readAsText(uploadedfile);
+    };
+
+    const handleDrop = (e: React.DragEvent) => {
+      e.preventDefault();
+      setDragOver(false);
+
+      const files = Array.from(e.dataTransfer.files);
+      console.log('Dropped files:', files[0]);
+      setUploadedFile(files[0] || null);
+    };
+
+    const handleDragOver = (e: React.DragEvent) => {
+      e.preventDefault();
+      setDragOver(true);
+    };
+
+    const handleDragLeave = () => setDragOver(false);
+
+    return (
+      <>
+        <Box
+          onDrop={handleDrop}
+          onDragOver={handleDragOver}
+          onDragLeave={handleDragLeave}
+          sx={theme => ({
+            position: 'relative',
+            borderRadius: 2,
+            p: 4,
+            textAlign: 'center',
+            border: '2px dashed',
+            borderColor: dragOver
+              ? 'primary.main'
+              : theme.palette.mode === 'dark'
+              ? 'grey.300'
+              : 'grey.800',
+            bgcolor: theme.palette.mode === 'dark' ? 'grey.800' : 'grey.300',
+            cursor: 'pointer',
+            transition: '0.2s',
+          })}
+        >
+          <input
+            type="file"
+            style={{
+              position: 'absolute',
+              top: 0,
+              left: 0,
+              width: '100%',
+              height: '100%',
+              opacity: 0,
+              cursor: 'pointer',
+            }}
+            onChange={e => setUploadedFile(e.target.files?.[0] || null)}
+          />
+
+          <Icon icon="mdi:upload" width="40" color="white" />
+          <Typography sx={{ m: 2 }}>
+            {dragOver
+              ? t('translation|Drop the file here...')
+              : t('translation|Select a file or drag and drop here')}
+          </Typography>
+          <Box
+            sx={{
+              p: 1,
+              bgcolor: 'yellow',
+              borderRadius: 1,
+              color: 'black',
+              fontWeight: 'bold',
+              '&:hover': { bgcolor: '#e6d600' },
+            }}
+            component="span"
+          >
+            {t('translation|Select File')}
+          </Box>
+        </Box>
+
+        {uploadedfile && (
+          <Box
+            sx={theme => ({
+              bgcolor: theme.palette.mode === 'dark' ? 'grey.800' : 'grey.300',
```

Comment:
please use a color from the theme, probably theme.palette.background.muted or another